opm/output/eclipse/summaryReader.hpp

```diff
@@ -0,0 +1,80 @@
+/*
++   Copyright 2016 Statoil ASA.
++
++   This file is part of the Open Porous Media project (OPM).
++
++   OPM is free software: you can redistribute it and/or modify it under the terms of the GNU General Public License as published by
++   the Free Software Foundation, either version 3 of the License, or
++   (at your option) any later version.
++
++   OPM is distributed in the hope that it will be useful,
++   but WITHOUT ANY WARRANTY; without even the implied warranty of
++   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
++   GNU General Public License for more details.
++
++   You should have received a copy of the GNU General Public License
++   along with OPM.  If not, see <http://www.gnu.org/licenses/>.
++   */
+
+
+#include <ert/ecl/ecl_sum.h>
+#include <iostream>
+#include <vector>
+#include <algorithm>
+#include <ert/util/stringlist.h>
+#include <ert/util/int_vector.h>
+#include <ert/util/bool_vector.h>
+#include <opm/common/ErrorMacros.hpp>
+
+
+struct Deviation {
+  double relative_deviation; 
+  double absolute_deviation; 
+};
+
+class SummaryReader {
+private:
+  ecl_sum_type* ecl_sum1 = nullptr;
+  ecl_sum_type* ecl_sum2 = nullptr;
+  ecl_sum_type* ecl_sum_file_short = nullptr;
+  ecl_sum_type * ecl_sum_file_long = nullptr;
+  stringlist_type* keys1 = stringlist_alloc_new();
+  stringlist_type* keys2 = stringlist_alloc_new(); 
+  stringlist_type * keys_short = stringlist_alloc_new();
+  stringlist_type * keys_long = stringlist_alloc_new();
+  double relative_tolerance_max = 0;
+  double relative_tolerance_median_max = 0;
+  double absolute_tolerance_max = 0;
+  std::vector<double> * referance_vec = nullptr;
+  std::vector<double> * ref_data_vec = nullptr;
+  std::vector<double> * checking_vec = nullptr;
+  std::vector<double> * check_data_vec = nullptr;
+public:
+  SummaryReader(){} 
+
+  SummaryReader(double relative_tolerance_max, double relative_tolerance_median_max):
+    relative_tolerance_max(relative_tolerance_max),
+    relative_tolerance_median_max(relative_tolerance_median_max)
+  {} 
+
+
+  bool open(const char* smspecFile1, const stringlist_type* unsmryFile1, const char* smspecFile2, const stringlist_type* unsmryFile2);
```

Comment:
It would be much easier to use this class it all ert details could be encapsulated inside this class. At this point, we are almost there, except for the use of stringlist_type. Also, the SMSPEC and UNSMRY file names are the same except for the case name, so perhaps it is sufficient to just pass the two case names (as const std::string&) to this method? Also, open() and close() should essentially be (or be called from) the constructor and destructor, and not require the user to manually deal with this.